### ISSUE TITLE
docs

### ISSUE THREAD

**Issue by jonschlinkert** [doowb/detect-file#2]
Just realized that the description (in package.json and on the repo) of this module actually says nothing about what the module actually does. This would probably be a lot more popular if people knew the value and purpose of this lib, and that it is able to successfully detect a path regardless of casing

**Comment by doowb** [comment]
Thanks! There aren't even any keywords! I'll get it updated.

**Comment by phated** [comment]
@doowb @jonschlinkert can we get this shaped-up for 1.0 (docs, project structure, etc)?  I also noticed that it depends on `fs-exists-sync` but that is no longer needed because they un-deprecated it.

I bring this up because it's used by `findup-sync` and I'm trying to update all the deps.

**Comment by jonschlinkert** [comment]
> can we get this shaped-up for 1.0 (docs, project structure, etc)?

alright, admit it... you like `update` don't you lol. jk

**Comment by doowb** [comment]
@phated I'll get this updated and bump to 1.0.

**Comment by doowb** [comment]
updated, added more documentation, and published as 1.0.

Updated the description, if either of you has a better description feel free to update it (or let me know and I'll update it) @phated I added you to the project.

**Comment by jonschlinkert** [comment]
> if either of you has a better description feel free to update it

It still doesn't say what it does. If this "resolves a file path", then why not just use `path.resolve()`? Meaning, not every OS handles casing the same way, which makes it more difficult to check if a file exists when the casing given by the user is different than the casing on the fs.  This takes care of that. Correct?

edit: @doowb is this the lib that uses a breadth-first search? If so, that's also relevant for the description.

**Comment by doowb** [comment]
Yeah, this takes care of finding a case sensitive file path on case sensitive OSes.

> uses a breadth-first search?

Yes, because it only searches the directory that the file name might exist in.

How's this?

> Consistently resolves a file path by detecting if the file path exists on case sensitive and case insensitive file systems. When necessary, uses a breadth-first search to detect and resolve the file path using the correct casing.

Then I'll add a "Why use this?" section in the readme that explains that different file systems may have different casings and that using `path.resolve()` doesn't take that into account.

**Comment by jonschlinkert** [comment]
that's great!

**Comment by jonschlinkert** [comment]
How's this

> Uses a breadth-first search, when necessary, to resolve the file path using the correct casing, and verify that the file actually exists. Returns consistent results regardless of case sensitivity of the file path and/or operating system.
